Libraries/CO.CDP.Authentication.Tests/ClaimServiceTests.cs

```diff
@@ -0,0 +1,41 @@
+using FluentAssertions;
+using Microsoft.AspNetCore.Http;
+using Moq;
+using System.Security.Claims;
+
+namespace CO.CDP.Authentication.Tests;
+public class ClaimServiceTests
+{
+    [Fact]
+    public void GetUserUrn_ShouldReturnUrn_WhenUserHasSubClaim()
+    {
+        var userUrn = "urn:fdc:gov.uk:2022:rynbwxUssDAcmU38U5gxd7dBfu9N7KFP9_nqDuZ66Hg";
+        var claims = new List<Claim> { new Claim("sub", userUrn) };
+        var identity = new ClaimsIdentity(claims, "TestAuthType");
+        var claimsPrincipal = new ClaimsPrincipal(identity);
+        var httpContext = new DefaultHttpContext { User = claimsPrincipal };
+        var httpContextAccessor = new Mock<IHttpContextAccessor>();
```

Comment:
This is a lot of setup code, that hinders the readability of the test. 

All that happens here, if I understood it well, is:

```csharp
var httpContextAccessor = GivenHttpContextWith(userUrn: userUrn)
```